Modules/resource.c

```diff
@@ -340,144 +340,136 @@ resource_methods[] = {
 /* Module initialization */
 
 
-static struct PyModuleDef resourcemodule = {
-    PyModuleDef_HEAD_INIT,
-    "resource",
-    NULL,
-    -1,
-    resource_methods,
-    NULL,
-    NULL,
-    NULL,
-    NULL
-};
-
-PyMODINIT_FUNC
-PyInit_resource(void)
+static int resource_exec(PyObject *module)
 {
-    PyObject *m, *v;
+#define ADD_INT(module, value)                         \
+    do {                                               \
+        if (PyModule_AddIntMacro(module, value) < 0) { \
```

Comment:
Oh. Calling PyModule_AddIntMacro() in a macro is not reliable. On macOS (with clang), it creates {"4": 4} dictionary: the value is converted to a string, but it's not the constant name. Fix:

```suggestion
        if (PyModule_AddIntConstant(module, #value, value) < 0) { \
```

You may align the other \ characters ;-)

Reply:
Thanks a million. I need a macOS to do some debug work ;)